albums/views.py

```diff
@@ -1,3 +1,122 @@
+        limit       = ALBUM_COUNT*page
+        offset      = limit - ALBUM_COUNT
+
+        if request.GET.get('genre'):
+            filter_set['genre__name'] = request.GET['genre']
+        
+        if request.GET.get('year'):
+            filter_set['release_date__year'] = int(request.GET['year'])  
+
+        search_string = request.GET.get('search')
+        search_key    = request.GET.get('search_key')
+                
+        if search_string:
+            if int(search_key) == 0:
+                query &= Q(title__icontains=search_string) | Q(description__icontains=search_string)
+            elif int(search_key) == 1:
+                query &= Q(title__icontains=search_string)
+            else:
+                query &= Q(description__icontains=search_string)
+        
+        albums = Album.objects.select_related('genre','release_type').prefetch_related('artist').filter(query,**filter_set)
+        
+        if not len(albums):
+            return JsonResponse({'MESSAGE':"PAGE_NOT_FOUND"}, status=404)
+        
+        page_count = math.ceil(albums.count()/ALBUM_COUNT)  
+        
+        data = {
+            "page" : page,
+            "page_count" : page_count,
+            "albums": [{
+                "album_id" : album.id,
+                "title" : album.title,
+                "artist" : [ artist.name for artist in album.artist.all()],
+                "release_type": album.release_type.name,
+                "release_date": album.release_date
+                
+            } for album in albums[offset:limit]]}   
+        
+        return JsonResponse(data,status=200)
+        
+class DetailPageView(View):
+    def get(self,request,album_pk):
+        try:
+            album = Album.objects.select_related(
+                'genre','release_type').prefetch_related(
+                'artist','song','play_link').get(id=album_pk)
+
+            previous_album  = Album.objects.prefetch_related('artist').filter(id__lt=album_pk).order_by('id').first()
+            next_album      = Album.objects.prefetch_related('artist').filter(id__gt=album.pk).order_by('id').first()
+            data = {
+                "id"                 : album.id,
+                "title"              : album.title,
+                "artist"             : [artist.name for artist in album.artist.all()],
+                "image_url"          : album.image_url,
+                "description"        : album.description,
+                "description_detail" : album.description_detail,
+                "release_date"       : album.release_date,
+                "release_type"       : album.release_type.name,
+                "genre"              : album.genre.name,
+                "song"               : [song.name for song in album.song.all()],
+                "playlinks"          : [{
+                    "link_name" : playlink.name,
+                    "link_url"  : AlbumPlayLink.objects.get(album=album.id,play_link=playlink.id).url
```

Comment:
여기서도 error 발생할 수 있겠네요.
```suggestion
                    "link_url"  : AlbumPlayLink.objects.get(album=album.id,play_link=playlink.id).url
```

Reply:
이 부분을 밑에 다음 COMMENT에 해주신 내용처럼 수정하여서, ```AlbumPlayLink.DoesNotExist```와 같은 에러는 발생하지 않도록 하였습니다 !
:+1: